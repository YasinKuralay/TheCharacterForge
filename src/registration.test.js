import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Registration from "./registration.js";

it("renders the Registration component without crashing", () => {
    const div = document.createElement("div");
    // The BrowserRouter is for eliminating the unnecessary error "Don't use Link outside Router" from Jest, which is caused by using an old version of the Router, and will be fixed
    ReactDOM.render(
        <BrowserRouter>
            <Registration />
        </BrowserRouter>,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the previous snapshot of the Registration Component", () => {
    const RegistrationComponentSnapshot = renderer
        .create(
            <BrowserRouter>
                <Registration />
            </BrowserRouter>
        )
        .toJSON();
    expect(RegistrationComponentSnapshot).toMatchSnapshot();
});
