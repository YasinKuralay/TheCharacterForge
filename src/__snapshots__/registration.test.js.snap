// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the Registration Component 1`] = `
<div
  id="register"
>
  <h2>
    Register
  </h2>
  <input
    autoComplete="off"
    name="firstname"
    onChange={[Function]}
    placeholder="firstname"
  />
  <input
    autoComplete="off"
    name="lastname"
    onChange={[Function]}
    placeholder="lastname"
  />
  <input
    autoComplete="off"
    name="email"
    onChange={[Function]}
    placeholder="email"
    type="email"
  />
  <input
    autoComplete="off"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
  />
  <button
    onClick={[Function]}
  >
    Register
  </button>
  <p
    id="ifYou"
  >
    If you already have an account, please
     
    <a
      href="/login"
      id="textDecoration"
      onClick={[Function]}
    >
      log in
    </a>
    .
  </p>
</div>
`;
